rclcpp/src/rclcpp/duration.cpp

```diff
@@ -214,4 +214,10 @@ Duration::nanoseconds() const
   return rcl_duration_.nanoseconds;
 }
 
+double
+Duration::seconds() const
+{
+  return std::chrono::duration<double>(std::chrono::nanoseconds(rcl_duration_.nanoseconds)).count();
```

Comment:
The current line in the patch matches the content of the same function in the `Time` class. I am not sure how the referenced issue applies here since the computation is not complex in any way?

Can you please propose the specific code you would prefer to implement this method.

Reply:
I recommend you use https://github.com/ros2/geometry2/blob/7dafb4d6c0c7025dfeeed3cef336113c62601dac/tf2/include/tf2/time.h#L74..L84 to make the computation more complex and more accurate.

Reply:
Got it now. The goal is to increase the precision.

It would be good to share that implementation somehow between the various places where we do convert integer nanoseconds into double seconds. That function should probably live in `rcutils` and would then be used here in `rclcpp` as well as `geometry2`. Since that exceeds the scope of this patch I would keep that for a separate set of PRs. (Also a test demonstrating that the precision is increased would be good for that function.)

Reply:
yeah, I imagine the precision is not expected to be perfect anyway, but the improvement is there for when we want it. can be an enhancement.